app/src/main/java/com/github/h3lp3rs/h3lp/MainPageActivity.kt

```diff
@@ -176,6 +187,38 @@ class MainPageActivity : AppCompatActivity() {
         return if (toggle.onOptionsItemSelected(item)) true else super.onOptionsItemSelected(item)
     }
 
+    override fun onRequestPermissionsResult(
+        requestCode: Int, permissions: Array<String>,
+        grantResults: IntArray
+    ) {
+        if (requestCode != LOCATION_PERMISSION_REQUEST_CODE) {
+            super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+            return
+        }
+        if (grantResults.isNotEmpty() && grantResults[0] == PERMISSION_GRANTED) {
+            // Do nothing yet
+        } else {
+            // Permission was denied.
+            // Display the missing permission error dialog when the fragments resume.
+            locPermissionDenied = true
+        }
+    }
+
+    override fun onResumeFragments() {
+        super.onResumeFragments()
+        if (locPermissionDenied) {
+            Toast.makeText(this, "Permission denied", Toast.LENGTH_SHORT).show()
```

Comment:
Be careful with using Toasts, I am not sure they work fine with Cirrus. Maybe check this in case of bugs.

Reply:
Oh good to know thanks ! In this case it shouldn't be a problem since the tests run with permission, but I will look carefully into it